### PR TITLE
[DO NOT MERGE] Fix issues in corehost that prevented hellomvc from working

### PR DESCRIPTION
**Pull request by pakrym**
1. Rid graph travers goes to infinite loop because `iter++` changes internal state of iterator but returns old one
2. We need to merge rid and rid-less assets instead of selecting only one.

Code is horrible.

/cc @schellap @anurse

### CODE REVIEW

File: src/corehost/cli/deps_format.cpp [hunk 1]
```diff
@@ -225,9 +230,21 @@ bool deps_json_t::load_portable(const json_value& json, const pal::string_t& tar
         return rid_assets.count(package) || non_rid_assets.count(package);
     };
     auto get_relpaths = [&rid_assets, &non_rid_assets](const pal::string_t& package, int type_index) -> const std::vector<pal::string_t>& {
-        return (rid_assets.count(package))
-            ? rid_assets[package].begin()->second[type_index]
-            : non_rid_assets[package][type_index];
+       // HACK:      v - memory leak
```
Comment (pakrym) [inline src/corehost/cli/deps_format.cpp]: :warning:

**Comment by schellap** [comment]
It just needs to be ++iter. But this is fine.

File: src/corehost/cli/deps_format.cpp (hunk 1 above)
Comment (schellap) [inline src/corehost/cli/deps_format.cpp]: Dont merge

Comment (pakrym) [inline src/corehost/cli/deps_format.cpp]: Wasn't going to, it's more to show issues and solutions to you.

Comment (schellap) [inline src/corehost/cli/deps_format.cpp]: The merge shoulf happpen already. Will TAL.

**Comment by anurse** [comment]
These are super-hacks that @pakrym and I bashed out to wrap up a few small things. This PR is illustrative, not an actual fix :).

**Comment by schellap** [comment]
Thanks for the testing it out, guys!

**Pull request closed without merging**